File: test/example/java/SocialNetworkTest.java
package com.vaticle.doc.test.example;

import com.vaticle.typedb.client.TypeDB;
import com.vaticle.typedb.client.api.TypeDBClient;
import com.vaticle.typedb.client.api.TypeDBSession;
import com.vaticle.typedb.client.api.TypeDBTransaction;
import com.vaticle.typeql.lang.TypeQL;
import com.vaticle.typeql.lang.common.TypeQLArg;
import com.vaticle.typeql.lang.query.TypeQLQuery;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.stream.Collectors;

import static com.vaticle.typeql.lang.TypeQL.*;

import org.junit.*;

public class SocialNetworkTest {

    @BeforeClass
    public static void loadSocialNetwork() {
        TypeDBClient client = TypeDB.coreClient("localhost:1729");
        client.databases().create("social_network");
        TypeDBSession session = client.session("social_network", TypeDBSession.Type.SCHEMA);
        TypeDBTransaction transaction = session.transaction(TypeDBTransaction.Type.WRITE);
        try {
            byte[] encoded = Files.readAllBytes(Paths.get("files/social-network/schema.tql"));
            String query = new String(encoded, StandardCharsets.UTF_8);
            transaction.query().define(TypeQL.parseQuery(query));
            transaction.commit();
            session.close();
            System.out.println("Loaded the social_network schema");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Test
    public void testSocialNetworkQuickstartQuery() { SocialNetworkQuickstartQuery.main(new String[]{}); }

    @Test
    public void testTypeDBQuickstartA() { TypeDBQuickstartA.main(new String[]{}); }

    @Test
    public void testTypeDBQuickstartB() { TypeDBQuickstartB.main(new String[]{}); }

    @Test
    public void testTypeDBQuickstartC() { TypeDBQuickstartC.main(new String[]{}); }

    @AfterClass
    public static void cleanSocialNetwork() {
        TypeDBClient client = TypeDB.coreClient("localhost:1729");
        client.databases().get("social_network").delete();
        System.out.println("Deleted the social_network database");
    }
}

File: test/example/python/social_network_test.py
from grakn.client import *
import unittest


class SocialNetworkTest(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        with open('files/social-network/schema.gql', 'r') as schema:
            define_query = schema.read()

            with Grakn.core_client() as client:
                if "social_network" in client.databases().all():
                    client.databases().get("social_network").delete()
                client.databases().create("social_network")
                with client.session("social_network", SessionType.SCHEMA) as session:
                    with session.transaction(TransactionType.WRITE) as transaction:
                        transaction.query().define(define_query)
                        transaction.commit()
                        print("Loaded the social_network schema")

    def test_social_network_quickstart_query(self):
        import social_network_quickstart_query

    def test_social_network_python_client_a(self):
        import social_network_python_client_a

    def test_social_network_python_client_b(self):
        import social_network_python_client_b

    def test_social_network_python_client_c(self):
        import social_network_python_client_c

    def test_social_network_python_client_d(self):
        import social_network_python_client_d


if __name__ == '__main__':
    unittest.main(verbosity=4)
